src/aws-healthomics-mcp-server/awslabs/aws_healthomics_mcp_server/utils/validation_utils.py

```diff
@@ -135,3 +135,39 @@ async def validate_container_registry_params(
             logger.error(error_message)
             await ctx.error(error_message)
             raise ValueError(error_message)
+
+
+async def validate_adhoc_s3_buckets(adhoc_buckets: Optional[List[str]]) -> List[str]:
+    """Validate adhoc S3 bucket paths and check access permissions.
+
+    This function validates bucket path formats and tests access permissions
+    for adhoc buckets that are not part of the standard configuration.
+
+    Args:
+        adhoc_buckets: List of S3 bucket paths to validate
+
+    Returns:
+        List of validated and accessible bucket paths
+
+    Raises:
+        ValueError: If no valid buckets are provided or accessible
+    """
+    if not adhoc_buckets:
+        return []
+
+    from awslabs.aws_healthomics_mcp_server.utils.s3_utils import validate_bucket_access
+
+    try:
+        # Use existing utility to validate bucket access
+        # This handles format validation, deduplication, and access testing
+        validated_buckets = validate_bucket_access(adhoc_buckets)
```

Comment:
This could lead to redundant AWS API calls when validating adhoc bucket access. May want to consider implementing caching to improve efficiency.